### DISCUSSION TITLE
Automatic sizing of `rightSection` for <Input>

### DISCUSSION THREAD

**Discussion by adamerose** [mantinedev/mantine#4747] [Feature requests]
The [rightSection](https://mantine.dev/core/text-input/#with-right-section) prop takes a React element but if you give it anything bigger than an icon the layout doesn't look good, it overlaps the text area and scrollbar and overflows out of the right side. You can fix it by setting `rightSectionWidth` to the size of your element, but this doesn't work if it has varying size and still overlaps the scrollbar. 

It would be nice if `rightSectionWidth` was determined automatically without needing the user to explicitly set it, so the contents of `rightSection` would always fit between the text to its left the the scrollbar to its right with no overflow/overlap. 

https://codesandbox.io/s/mantine-sandbox-forked-ws68n5

<img width="496" alt="image" src="https://github.com/mantinedev/mantine/assets/10884874/1e0e0ada-9a0e-46bc-91e0-238754bbcf7b">

**Comment by daniel-aziz** [comment]
Any update on this? 
its much needed.

**Comment by Pirulax** [comment]
Yeah would be nice to have something to accomplish this
